src/app/resources/view-resources/resources-view.component.ts

```diff
@@ -89,6 +102,66 @@ export class ResourcesViewComponent implements OnInit, OnDestroy {
       this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
     }
     this.couchSrc = this.urlPrefix + resource._id + '/' + filename;
+    // resource rating
+    this.couchService.get('ratings/_all_docs?include_docs=true')

```

Comment:
You should do a mango query to find the specific ratings for this resource rather then fetch all resources & `map` and `filter` over each value.

Reply:
Used mango query